### PR TITLE
Milestone 10: Authorization rules

### PR DESCRIPTION
**Pull request by skabeo**
In this pull request I;

- Added cancan gem
- Added authorizations on a post for a user which included:
  - Users can read and only delete their post
  - Admins can read and delete every post
- Added a role column to the users table
- Added a delete button in the `app/views/posts/show.html.erb` file to allow identified users delete their post
- Added a delete comment path to the route
- Created a delete method in the comments controller to:
  - Give read and delete access to comments authors
  - Give read and delete access to admins
- Added a `delete comment` button only visible for people authorized to delete a comment

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    can(:read, Comment, user:)
+    can :destroy, Comment.where(author_id: user.id)
+
+    return unless user.role == 'admin'
+
+    can(:read, Comment, user:)
+    # can :destroy, Comment.where(author_id: user.id)
+    can :destroy, Post
+    can :destroy, Comment
```
Comment (brytebee) [inline app/models/ability.rb]: - [ ] **Optional:**

This lines of code can be replaced with 1.

`can :manage, :all`

File: app/controllers/posts_controller.rb
Comment (brytebee) [inline app/controllers/posts_controller.rb:1]: At the top of your `Post` class, kindly add the line below as [a helper for `cancancan`.](https://github.com/CanCanCommunity/cancancan#controller-helpers)

`load_and_authorize_resource`

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @skabeo & @ikennarichard 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great integrating the `cancancan` gem for authorization.✔️
- Awesome defining the role attribute of a class and setting permission for the roles in the ability model.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @skabeo, @ikennarichard 🙋‍♀️,  

Excellent work! 💯 
You have done a great job adding the Blog app authorization rules. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞 

✔️  Good PR title and description
✔️  Users are only authorized to delete posts and comments they own
✔️  The `role` column exists in the user's table
✔️  There's a CanCanCan `ability` class that defines the set of rules for user-post interactions
✔️  Project has **CanCan** Gem in **Gemfile**

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**